File: document/src/vespa/document/bucket/bucket.cpp
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "bucket.h"
#include <vespa/vespalib/stllike/asciistream.h>

namespace document {

Bucket::Bucket() noexcept
    : _bucketSpace(BucketSpace::invalid()),
      _bucketId()
{
}

vespalib::string Bucket::toString() const
{
    vespalib::asciistream os;
    os << *this;
    return os.str();
}

vespalib::asciistream& operator<<(vespalib::asciistream& os, const Bucket& id)
{
    return os << "Bucket(" << id.getBucketSpace() << ", " << id.getBucketId() << ")";
}

std::ostream& operator<<(std::ostream& os, const Bucket& id)
{
    return os << id.toString();
}

}

File: persistence/src/main/java/com/yahoo/persistence/spi/Bucket.java
// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.persistence.spi;

import com.yahoo.document.BucketId;

/**
 * @author thomasg
 */
public class Bucket {
    BucketId bucketId;
    short partitionId;

    /**
     * @param partition The partition (i.e. disk) where the bucket is located
     * @param bucketId The bucket id of the bucket
     */
    public Bucket(short partition, BucketId bucketId) {
        this.partitionId = partition;
        this.bucketId = bucketId;
    }

    public BucketId getBucketId() { return bucketId; }

    public short getPartitionId() { return partitionId; }

    @Override
    public String toString() {
        return partitionId + "/" + bucketId;
    }
}
